core/src/contracts/stablex_contract.rs

```diff
@@ -377,6 +367,95 @@ async fn get_block(
         .ok_or_else(|| anyhow!("block {:?} is missing", block_number))
 }
 
+trait BatchIdGetter {
```

Comment:
I would like to put the new traits and the test into their own file but depends on what others think.

Reply:
+1 for own file (`stablex_contract.rs` is huge as it is). Also, I personally prefer calling traits with ["gerunds"](https://en.wikipedia.org/wiki/Gerund) (`BatchIdGetting` in this case). This is because the trait itself doesn't actually perform the action but it describes what implementations of it can do (read: "this is the trait for getting batch ids").

Reply:
I also think this is a good idea, it keeps the logic, tests, etc. associated with the new trait in its own module so that it is easier to find, and makes the modules contain more immediately relevant information.

Reply:
I like the suggestion on changing the name as well, especially since pretty much all of our traits in this codebase conform to this naming, making them easy to identify by name. But maybe `BatchIdRetrieving` or `BatchIdSearching` as `BatchIdGetting` sounds a little funny to my ears.